compiler/ilgen/OMRMethodBuilder.hpp

```diff
@@ -351,7 +351,7 @@ class MethodBuilder : public TR::IlBuilder
    TR::IlType                ** _cachedParameterTypes;
    const char                * _definingFile;
    char                        _definingLine[MAX_LINE_NUM_LEN];
-   TR::IlType                * _cachedParameterTypesArray[10];
+   TR::IlType                * _cachedParameterTypesArray[18];
```

Comment:
This seems less than ideal. There should be a better way of doing this than hardcoding a constant array length and asserting that the number of parameters is less than 18. We should use a dynamic data structure and allow as many parameters as the user wants (within reason).

Reply:
Agreed, but you shouldn't have to fix that here. Created #4748 .